catroid/src/test/java/org/catrobat/catroid/test/roboelectric/formulaeditor/FormulaEditorEditTextGenericTest.java

```diff
@@ -0,0 +1,126 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.ParameterizedRobolectricTestRunner;
+import org.robolectric.Robolectric;
+import org.robolectric.android.controller.ActivityController;
+
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(ParameterizedRobolectricTestRunner.class)
+public class FormulaEditorEditTextGenericTest {
+
+	@ParameterizedRobolectricTestRunner.Parameters(name = "{0}" + "-Test")
+	public static Iterable<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+				{"button_0", R.id.formula_editor_keyboard_0, "0 "},
+				{"button_1", R.id.formula_editor_keyboard_1, "1 "},
+				{"button_2", R.id.formula_editor_keyboard_2, "2 "},
+				{"button_3", R.id.formula_editor_keyboard_3, "3 "},
+				{"button_4", R.id.formula_editor_keyboard_4, "4 "},
+				{"button_5", R.id.formula_editor_keyboard_5, "5 "},
+				{"button_6", R.id.formula_editor_keyboard_6, "6 "},
+				{"button_7", R.id.formula_editor_keyboard_7, "7 "},
+				{"button_8", R.id.formula_editor_keyboard_8, "8 "},
+				{"button_9", R.id.formula_editor_keyboard_9, "9 "},
+				{"button_decimal_mark", R.id.formula_editor_keyboard_decimal_mark, "0. "},
+				{"button_plus", R.id.formula_editor_keyboard_plus, "+ "},
+				{"button_minus", R.id.formula_editor_keyboard_minus, "- "},
+				{"button_mult", R.id.formula_editor_keyboard_mult, "× "},
+				{"button_divide", R.id.formula_editor_keyboard_divide, "÷ "},
+				{"button_bracket_open", R.id.formula_editor_keyboard_bracket_open, "( "},
+				{"button_bracket_close", R.id.formula_editor_keyboard_bracket_close, ") "}
+		});
+	}
+
+	@IdRes
+	private int formulaKeyboardItemId;
+	private String expectedFormulaText;
+
+	private SpriteActivity activity;
+	private FormulaBrick brick;
+
+	public FormulaEditorEditTextGenericTest(String name, @IdRes int formulaKeyboardItemId, String expectedFormulaText) {
+		this.expectedFormulaText = expectedFormulaText;
+		this.formulaKeyboardItemId = formulaKeyboardItemId;
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		ActivityController<SpriteActivity> activityController = Robolectric.buildActivity(SpriteActivity.class);
+		activity = activityController.get();
+		brick = new SetXBrick();
+		createProject(activity, brick);
+		activityController.create().resume();
+		View brickView = brick.getView(activity);
+		TextView brickFormulaTextView = brickView.findViewById(R.id.brick_set_x_edit_text);
+		brick.onClick(brickFormulaTextView);
+	}
+
+	@Test
+	public void openFormulaEditorTest() {
+		MotionEvent touchDown = MotionEvent.obtain(SystemClock.uptimeMillis(), SystemClock.uptimeMillis(),
+				MotionEvent.ACTION_DOWN, 1, 1, 0);
+		View formulaKeyboardItemView = activity.findViewById(formulaKeyboardItemId);
+		formulaKeyboardItemView.dispatchTouchEvent(touchDown);
+		EditText formulaEditField = activity.findViewById(R.id.formula_editor_edit_field);
+		assertEquals(expectedFormulaText, formulaEditField.getText().toString());
```

Comment:
use:
formulaEditField.getText().toString().trim() and remove the trailing whitespaces from the parameters

Reply:
Done, thanks!